lib/DataIntegrityProof.js

```diff
@@ -164,19 +166,23 @@ export class DataIntegrityProof extends LinkedDataProof {
 
     // set default `now` date if not given in `proof` or `options`
     let date = this.date;
-    if(proof.created === undefined && date === undefined) {
-      date = new Date();
-    }
+    if(date === null) {
+    } else {
```

Comment:
Perhaps rewrite as `if(date !== null)` or add a comment in the empty clause about omitting the `created` property in the `null` case.